graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/processors/ConfigurationStateUpdater.java

```diff
@@ -104,6 +107,17 @@ private synchronized PipelineInterpreter.State reloadAndSave() {
         return newState;
     }
 
+    private PipelineInterpreter.State reloadAndSave(RulesChangedEvent event) {
+        final PipelineInterpreter.State state = reloadAndSave();
+        metadataListener.handleRuleChanges(event, state, pipelineResolver, pipelineMetricRegistry);
+        return state;
+    }
+
+    private PipelineInterpreter.State reloadAndSave(PipelinesChangedEvent event) {
+        final PipelineInterpreter.State state = reloadAndSave();
+        metadataListener.handlePipelineChanges(event, state, pipelineResolver, pipelineMetricRegistry);
+        return state;
+    }
```

Comment:
These will run on every node for every change, right?
If that's correct we should probably only run it on the leader node, otherwise we will end up doing the same work on every node in the cluster, but in an unspecified order, which might lead to slightly different state depending on time.
Probably not the end of the world for this use case, but if we can avoid it, that would be good.
Even better: we only recalculate the metadata on the node that performed the REST calls, but that might get tricky if there are other sources of changes, so I think the leader node is probably be the safest approach.